paddle/fluid/operators/scale_op.cc

```diff
@@ -52,6 +55,21 @@ Multiply the input tensor with a float scalar to scale the input tensor.
   }
 };
 
+class ScaleOpVarTypeInference : public framework::VarTypeInference {
+ public:
+  void operator()(const framework::OpDesc &op_desc,
+                  framework::BlockDesc *block) const override {
+    auto &in_var_name = op_desc.Input("X").front();
+    auto &in_var = detail::Ref(block->FindVarRecursive(in_var_name));
+
+    auto out_var_name = op_desc.Output("Out").front();
+    auto &out_var = block->FindRecursiveOrCreateVar(out_var_name);
```

Comment:
Should the output not to "find recursive"? because the output may have the same name as in parent blocks. Should act like a "program".

Reply:
Done